### PR TITLE
Restructured a bunch of game logic and fixed some turn order bugs.

### PR DESCRIPTION
**Pull request by Floxnu**

### CODE REVIEW

File: src/main/java/ca/ciccc/silverBullet/gameBoard/GridBoard.java [hunk 1]
```diff
@@ -252,5 +252,14 @@ public boolean tryShoot (Player playerShooting){
...
     }
+
+    public boolean areAllFull(){
+        for (Player p : players){
+            if(!p.isActionsFull()){
+                return false;
+            }
+        }
+        return true;
+    }
```
Comment (nator333) [inline src/main/java/ca/ciccc/silverBullet/gameBoard/GridBoard.java]: Easy way to do this
```
return this.players.stream().allMatch(Player:: isActionsFull)
```

File: src/main/java/ca/ciccc/silverBullet/gameBoard/GameScene.java [hunk 2]
```diff
@@ -63,54 +57,60 @@ public void boardUpdate(){
...
+                case T:
+                    gameBoard.players.get(controllingPlayer).addAction(PlayerAction.WAIT);
+                    break;
+            }
         }
-        if(testFirstPlayer.isActionsFull()){
+        if(!gameBoard.areAllFull() && gameBoard.players.get(controllingPlayer).isActionsFull()){
             controllingPlayer++;
         }
+        if(key.equals(KeyCode.SPACE)){
```
Comment (nator333) [inline src/main/java/ca/ciccc/silverBullet/gameBoard/GameScene.java]: if key was null, this causes NPE.
Basically, code bellow is always better

```
if(KeyCode.SPACE.equals(key)){}
```

Comment (Floxnu) [reply src/main/java/ca/ciccc/silverBullet/gameBoard/GameScene.java]: Thank you, changed.

File: src/main/java/ca/ciccc/silverBullet/gameBoard/GridBoard.java (hunk 1 above)
Comment (Floxnu) [reply src/main/java/ca/ciccc/silverBullet/gameBoard/GridBoard.java]: Cool. Changed as well.

Comment (nator333) [reply src/main/java/ca/ciccc/silverBullet/gameBoard/GridBoard.java]: 4b1540ace1d5dc853e226e6ad0d35fdbc42b65ac

Comment (Floxnu) [reply src/main/java/ca/ciccc/silverBullet/gameBoard/GridBoard.java]: 00e25cde836265381f43360a33c9c9eb5646139d

**Pull request merged**